### ISSUE TITLE
False positive for EF0004 Always tag the version of an image explicitly

### ISSUE THREAD

**Issue by tymonx** [buddy-works/dockerfile-linter#18]
Example Dockerfile:

```Dockerfile
FROM alpine:3.15.0 AS stage-A

FROM stage-A
```

Testing:

```plaintext
docker run -it --rm --workdir "$(pwd)" --volume "$(pwd):$(pwd):ro,z" --user "$(id -u):$(id -g)" buddy/dockerfile-linter:1.1.5 linter --file Dockerfile
```

Output:

```plaintext
Line 3:
   /EF0004/ Always tag the version of an image explicitly
```

This will always fail for local multi stages when using characters like `-` dash for intermediate image name. According to [docker tag](https://docs.docker.com/engine/reference/commandline/tag/):

```plaintext
Name components may contain lowercase letters, digits and separators. A separator is defined as a period, one or two underscores, or one or more dashes. A name component may not start or end with a separator.

A tag name must be valid ASCII and may contain lowercase and uppercase letters, digits, underscores, periods and dashes. A tag name may not start with a period or a dash and may contain a maximum of 128 characters.
```

This `Dockerfile` works fine with Docke CE, Moby and Podman. False positive error is always triggered for every local multi stage that contains characters like `-` dashes.

Expected behavior: no output with status code 0 (success).

**Comment by BartoszWrobel591** [comment]
Thank you for report, fixed in version 1.2.1.
